### ISSUE TITLE
Active24 REST API v2

### ISSUE THREAD

**Issue by kurm4n** [jahudka/certbot-dns-active24#13]
Hi,

just to be sure...

Are you planning to make modifications to maintain the functionality for the new version of Active24 API??

BR,

Josef

**Comment by jahudka** [comment]
Hi.. yeah. I mean I'll have to do it, eventually; I need it myself, after all. But it's been a while since I made this plugin; I'll have to look into whether there are any significant updates to the Certbot plugin architecture to figure out how exactly to go about upgrading, and I haven't had a lot of time lately. I've actually reached out to Active24 to ask if they maybe want to take over the maintenance of this plugin, since it wouldn't cost them much and it would be a good service to offer their customers, but they aren't interested. It'd be great if anybody else wanted to help out, because I'm not really a Python developer, so it's pretty tedious for me, but alas, in all the years this plugin existed, nobody expressed interest in contributing..

**Comment by kurm4n** [comment]
Hi!

Thanks for the reply. I am not a Python developer neither but maybe I can be helpful. I will try to look into it also a little bit.

Anyway thanks for making this plugin ! For now it is a very very very handy and helpful tool for me.

**Comment by jahudka** [comment]
Hi, just an update: I have started looking into this, I already have most of the code ready, but unfortunately I'm running into a pretty bad issue: the new API requires you to use the _service ID_ to access and manage DNS records, as opposed to the domain name like the old API used to - and there doesn't appear to be an endpoint which would list all services in an account or anything like that. The only way I found out to obtain the service ID for a particular domain is to copy it from an URL in the new Active24 Admin Dashboard, which sux. I've reached out to Active24 to ask for advice; but it may end up being the case that you need to specify the service ID manually when requesting certificates. I'd probably either add a command line option for that, or (if the Certbot plugin architecture doesn't preserve command line options for renewals, which I'm currently unsure about) as a map of domain name -> service ID in the config file. But it would still be much better if this could be done automatically of course..

**Comment by kurm4n** [comment]
Hey!

Kinda great news kinda not. But anyway - once we are setting up the cerbot at the beginning there is no such pain to add the _service ID_ to the config file, isn't it?

So I guess that it is fine once it is set once at the beginning.......

Thanks for working on that!

**Comment by jahudka** [comment]
Hi, apparently there is a "v1" REST API which is _different_ from what this plugin was using all this time, and that API has an endpoint to get a list of services for the current user. So you may now feel free to test the just released Version Two of the Certbot Active24 DNS Authenticator Plugin :-)

**Comment by kurm4n** [comment]
Hey! That's terrific! 

Thank you!

Is there any way how can I support you?
A little gift or something like that?

I would like to somehow pay back your effort.

BR,

Josef

**Comment by kurm4n** [comment]
Hey,

just checked that and tried. Everything is fine and working but there are two things I have noticed.

Running on Ubuntu Server the plugin doesn't name "certbot-dns-active24:dns-active24" in certbot but simply "dns-active24" (in setting up the request for the certificate.

And the second one is that in the active24.ini  "certbot_dns_active24:dns_active24_api_key" is not recognized instead I tried simply "dns_active24_api_key" and that is working fine.

ONCE AGAIN THANK YOU!!

BR,

Josef

**Comment by jahudka** [comment]
Hey, thanks for testing, glad it works for you!

> Is there any way how can I support you?

Nah, no worries ;-) thanks for offering though! If you ever come across somebody who you think might be a good fit for a second maintainer, that'd be awesome, but other than that, I mostly made this plugin for myself and it's pretty low-maintenance anyway.

Regarding the "certbot-dns-active24:dns-active24" vs "dns-active24" on the command line and the analogous config keys, funny thing is, it's now working for you exactly as it was originally expected and intended to, but people reported they had the problem the other way around - there's a closed issue on here about it - I think I even read that _all_ 3rd party Certbot plugins have this same issue and that it's on Certbot's side and not easy to fix without breaking stuff for too many people. Hence why, a few years ago, I changed the docs to include the `certbot[-_]dns[-_]active24:` prefixes. Curious to know which version of Certbot itself you're running, and whether it's from PyPI or APT? I'd like to pinpoint what exactly the difference is between installations which do require the prefixes and ones which don't, so I can update the docs accordingly. Thanks!

**Comment by jahudka** [comment]
Looks like using plugins without the prefixes was enabled by [this PR](https://github.com/certbot/certbot/pull/8131) in Certbot, which was merged all the way back in July 2020, making it probably part of the 1.7.0 stable release of Certbot; that PR allowed using both the (deprecated) prefixed names and the (correct) unprefixed ones; I'm not sure when they finally stopped supporting the prefixed versions, but it must be a while now. Anyway, looks like unless you're running an _ancient_ version of Certbot, unprefixed is the way to go. I'll update the readme. Thanks again for testing this, obviously I tested it locally and I found out the same thing, but I thought it's just a quirk of the local development environment.
